materialize-sql/validate.go

```diff
@@ -100,13 +100,16 @@ func ValidateMatchesExisting(resource Resource, existing *pf.MaterializationSpec
 	var constraints = make(map[string]*pm.Constraint)
```

Comment:
Nit: Might update the comment for this function where it says `...will explicitly require all fields that are currently materialized...`

Reply:
This actually reminded me to add an additional check here to validate that keys from the existing materialization are present in the new materialization as well so we don't allow the validate step to pass only to error on ApplyUpsert